File: app/src/main/java/com/angogasapps/myfamily/objects/MessageNotification.java
package com.angogasapps.myfamily.objects;

import android.app.Notification;
import android.app.PendingIntent;
import android.content.Context;
import android.content.Intent;

import androidx.core.app.NotificationCompat;

import com.angogasapps.myfamily.R;
import com.angogasapps.myfamily.database.UserDao;
import com.angogasapps.myfamily.ui.screens.chat.ChatActivity;

import io.reactivex.annotations.NonNull;

import static com.angogasapps.myfamily.app.AppNotificationManager.CHANNEL_ID;
import static com.angogasapps.myfamily.firebase.FirebaseVarsAndConsts.TYPE_IMAGE_MESSAGE;
import static com.angogasapps.myfamily.firebase.FirebaseVarsAndConsts.TYPE_TEXT_MESSAGE;
import static com.angogasapps.myfamily.firebase.FirebaseVarsAndConsts.TYPE_VOICE_MESSAGE;


public class MessageNotification {

    public static Notification build(Context context, Message message, UserDao userDao){
        return createNotification(context, message, userDao);
    }

    public static Notification build(Context context, Message message, String userName){
        return createNotification(context, message, userName);
    }

    private static Notification createNotification(Context context, Message message, UserDao userDao){
        User user = userDao.getById(message.getFrom());

        return createNotification(context, message, user == null ? "___" : user.getName());
    }

    private static Notification createNotification(Context context, Message message, String userName){
        NotificationCompat.Builder builder =
                new NotificationCompat.Builder(context, CHANNEL_ID)
                        .setContentTitle(context.getString(R.string.my_family))
                        .setContentText(getTextToMessageNotification(context, message, userName))
                        .setSmallIcon(R.drawable.default_user_photo);


        Intent intent = new Intent(context, ChatActivity.class);
        PendingIntent pIntent = PendingIntent.getActivity(context, 0, intent, PendingIntent.FLAG_UPDATE_CURRENT);

        builder.setContentIntent(pIntent)
                .setAutoCancel(true)
                .setPriority(NotificationCompat.PRIORITY_MAX)
                .setOngoing(false);

        return builder.build();

    }

    private static String getTextToMessageNotification(Context context, Message message, String userName){
        StringBuilder string = new StringBuilder();


        string.append(userName);
        string.append(": ");


        if (message.getType().equals(TYPE_TEXT_MESSAGE))
            string.append(message.getValue());
        else if (message.getType().equals(TYPE_IMAGE_MESSAGE))
            string.append("\uD83D\uDCF7" + " ").append(context.getString(R.string.photo));
        else if (message.getType().equals(TYPE_VOICE_MESSAGE))
            string.append("\uD83C\uDFA4" + " ").append(context.getString(R.string.voice));

        return string.toString();
    }

}

File: app/src/main/java/com/angogasapps/myfamily/models/MessageNotification.kt
package com.angogasapps.myfamily.models

import android.app.Notification
import com.angogasapps.myfamily.database.UserDao
import com.angogasapps.myfamily.app.AppNotificationManager
import com.angogasapps.myfamily.R
import android.content.Intent
import com.angogasapps.myfamily.ui.screens.chat.ChatActivity
import android.app.PendingIntent
import android.content.Context
import androidx.core.app.NotificationCompat
import com.angogasapps.myfamily.firebase.*
import java.lang.StringBuilder

object MessageNotification {
    fun build(context: Context, message: Message, userDao: UserDao): Notification {
        return createNotification(context, message, userDao)
    }

    fun build(context: Context, message: Message, userName: String): Notification {
        return createNotification(context, message, userName)
    }

    private fun createNotification(
        context: Context,
        message: Message,
        userDao: UserDao
    ): Notification {
        val user = userDao.getById(message.from)
        return createNotification(context, message, if (user == null) "___" else user.name)
    }

    private fun createNotification(
        context: Context,
        message: Message,
        userName: String
    ): Notification {
        val builder = NotificationCompat.Builder(context, AppNotificationManager.CHANNEL_ID)
            .setContentTitle(context.getString(R.string.my_family))
            .setContentText(getTextToMessageNotification(context, message, userName))
            .setSmallIcon(R.drawable.default_user_photo)
        val intent = Intent(context, ChatActivity::class.java)
        val pIntent =
            PendingIntent.getActivity(context, 0, intent, PendingIntent.FLAG_UPDATE_CURRENT)
        builder.setContentIntent(pIntent)
            .setAutoCancel(true)
            .setPriority(NotificationCompat.PRIORITY_MAX)
            .setOngoing(false)
        return builder.build()
    }

    private fun getTextToMessageNotification(
        context: Context,
        message: Message,
        userName: String
    ): String {
        val string = StringBuilder()
        string.append(userName)
        string.append(": ")
        when (message.type) {
            TYPE_TEXT_MESSAGE -> string.append(message.value)
            TYPE_IMAGE_MESSAGE -> string.append("\uD83D\uDCF7" + " ")
                .append(context.getString(R.string.photo))
            TYPE_VOICE_MESSAGE -> string.append(
                "\uD83C\uDFA4" + " "
            ).append(context.getString(R.string.voice))
        }
        return string.toString()
    }
}